### PR TITLE
OpenAPI specification to read and delete pubkey

### PR DESCRIPTION
**Pull request by rubhanazeem**
Specifications for `GET,DELETE 'source/:project/_pubkey'`

### CODE REVIEW

**Comment by rubhanazeem** [comment]
[Sources - Projects (L)](https://trello.com/c/MSXY9t4w/1986-sources-projects-l)

File: src/api/public/apidocs-new/paths/source_project_name_pubkey.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      description: OK. The request has succeeded.
+      content:
+        text/plain:
+          schema:
+            type: string
+            example: gpgkey
+    '401':
+      $ref: '../components/responses/unauthorized.yaml'
+    '404':
+      $ref: '../components/responses/unknown_project.yaml'
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/source_project_name_pubkey.yaml:21]: I will add here the case where the pubkey is not found.

This is what I have teststed:
```
> curl -u enavarro_suse https://api-test.opensuse.org/source/home:enavarro_suse/_pubkey
Enter host password for user 'enavarro_suse':
<status code="404">
  <summary>home:enavarro_suse: no pubkey available</summary>
  <details>404 home:enavarro_suse: no pubkey available</details>
</status>
```

Adding this last case should not imply removing the already documented case of unknown project.

File: src/api/public/apidocs-new/paths/source_project_name_pubkey.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  parameters:
+    - $ref: '../components/parameters/project_name.yaml'
+  responses:
+    '200':
+      description: OK. The request has succeeded.
+      content:
+        text/plain:
+          schema:
+            type: string
+            example: gpgkey
```
Comment (eduardoj) [inline src/api/public/apidocs-new/paths/source_project_name_pubkey.yaml:17]:
```suggestion
            example: |
              -----BEGIN PGP PUBLIC KEY BLOCK-----
              Version: GnuPG v1.4.5 (GNU/Linux)

              mQENBFfX+W4BCAC89VZEUt8O/o2rXnFoNrW0nsZoaHZX5Kjtzwuns5/UwzBffY/h
              [...]
              -----END PGP PUBLIC KEY BLOCK-----
```

Even not being a full PGP public key, I would change the example of the text to something more similar.

**Comment by rubhanazeem** [comment]
Rebased with upstream. No need to review again

**Pull request merged**